browser/src/app/TilesMiddleware.ts

```diff
@@ -1201,7 +1076,7 @@ class TileManager {
 		)
 			this.emptyTilesCount -= 1;
 
-		this.reclaimTileCanvasMemory(tile);
+		this.reclaimTileBitmapMemory(tile);
```

Comment:
Maybe we can add a guard to this function, preventing tiles from being removed if they have pending deltas?

Reply:
I mean a guard to "removeTile".

Reply:
I think your comments are enough to make me think that we should key these rather than relying on order. I don't think there are many/any cases where tiles will disappear, but if they did I think behaviour would be unpredictable and better to guard against it.

Reply:
There is one case i could think of. You are calling Promise.all also for non-worker paths. That is an async call.

Other than that, yes a key for images would clear these concerns. Anyway, up to you, thanks :)